api/routes/database.py

```
@@ -36,27 +40,38 @@ async def connect_database(request: Request, db_request: DatabaseConnectionReque
         if url.startswith("postgres://") or url.startswith("postgresql://"):
             try:
                 # Attempt to connect/load using the PostgreSQL loader
                 success, result = PostgresLoader.load(request.state.user_id, url)
             except (ValueError, ConnectionError) as e:
                 logging.error("PostgreSQL connection error: %s", str(e))
-                raise HTTPException(status_code=500, detail="Failed to connect to PostgreSQL database")
-        
+                raise HTTPException(
+                    status_code=500,
+                    detail="Failed to connect to PostgreSQL database",
+                )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)